schema/components.json

```diff
@@ -565,6 +565,106 @@
           ]
         }
       ]
+    },
+    "PublicListing": {
+      "type": "object",
+      "title": "Public listing",
+      "description": "Details of a publicly listed company, its securities (shares and other tradable financial instruments related to the entity), and related regulatory filings.",
+      "required": [
+        "securitiesListings",
+        "hasPublicListing"
+      ],
+      "minProperties": 1,
+      "properties": {
+        "hasPublicListing": {
+          "type": "boolean",
+          "title": "Has public listing",
+          "description": "This value MUST be true if the entity is known to be a publicly listed company."
+        },
+        "companyFilingsURLs": {
+          "type": "array",
+          "title": "Company filings URLs",
+          "description": "URL or URLs where regulatory filings related to major holdings can be retrieved. URLs may point to pages maintained by regulatory bodies, stock exchanges or by the company itself.",
+          "items": {
+            "type": "string"
+          },
+          "format": "uri"
+        },
+        "securitiesListings": {
+          "type": "array",
+          "title": "Securities listings",
+          "description": "Details of the entity’s securities and the public exchanges and markets on which they are traded. All equity securities SHOULD BE listed here, plus any other securities from which beneficial ownership might be derived. Where a security is traded on more than one market, there SHOULD BE an entry for each market (or market segment).",
+          "items": {
+            "title": "Securities listing",
+            "description": "Details of a security and the exchange, trading platform or market segment on which it is traded.",
+            "$ref": "#/definitions/SecuritiesListing"
+          }
+        }
+      }
+    },
+    "SecuritiesListing": {
+      "type": "object",
+      "title": "Securities listing",
+      "description": "Details of a security and the exchange, trading platform or market segment on which it is traded.",
```

Comment:
I think this is a little vague. From the schema I can see that  "stockExchangeJurisdiction", "security" and "stockExchangeName" are required and so they MUST be included. I'm not sure from this description when the trading platform and/or market segment SHOULD be included.

Reply:
To address this we could add to the two child MIC field descriptions the sentence:

"MIC values SHOULD be published where they are known by the publisher."

But I wonder whether this goes without saying for all non-required fields, @siwhitehouse?

Reply:
It's really just in the context of this description that I think we could be clearer. I was using MUST and SHOULD to explain my concern, rather than suggest we use them explicitly  in the description. Sorry if that wasn't clear. I was thinking that something like

"Details of a security and the exchange, which may include the trading platform and/or market segment on which it is traded."

might work.

Reply:
@siwhitehouse - I'm not sure that works. I wonder if the description of stockExchangeName, as it stands, helps.

"The name of the exchange, market or trading platform on which the security is traded. If the security is traded on a segment of the exchange, then the name SHOULD include both elements. For example, 'London Stock Exchange - MTF'."

My understanding is that stock exchanges are now considered just one type of regulated trading platform. We're using the term 'stockExchangeName' since it's more meaningful to people, but the description makes clear that it should be the name of whatever the regulated trading platform is (even if it's not a traditional 'stock exchange'). The MIC fields give additional useful identifiers for the 'exchange, market or trading platform on which the security is traded'.

Reply:
@kd-ods My apologies, I don't think I'm being very clear and it feels like you are analysing this more than I am. 

I am reading

> Details of a security and the exchange, trading platform or market segment on which it is traded.

to mean that **one, and only one** from the exchange, the trading platform or the market segment are included. That's all I am suggesting that we try and resolve.

Reply:
Ahhhhhh. Understood. Would simplifying the description to the following work, then:

"Details of a security and the market on which it is traded."

Reply:
@kindly - that string has now been updated in the schema-generating sheet. It's ready to pull in.